modules/exploits/windows/smb/ms17_010_eternalblue_win8.py

```diff
@@ -673,7 +673,12 @@ def exploit(args):
     module.log('shellcode size: {:d}'.format(len(sc)))
     module.log('numGroomConn: {:d}'.format(numGroomConn))
 
-    _exploit(args['RHOST'], rport, feaList, sc, numGroomConn, smbuser, smbpass)
+    try:
+        _exploit(args['RHOST'], rport, feaList, sc, numGroomConn, smbuser, smbpass)
+    # XXX: Catch everything until we know better
+    except Exception as e:
```

Comment:
Haven't found anything better, and @acammack-r7 is pointing out that the Impacket exceptions inherit from `Exception` all the same. :/

Reply:
Hey man, i'm facing problem till 
at first it show like this upon 'run' :
msf5 exploit(windows/smb/ms17_010_eternalblue_win8) > run

[*] Started reverse TCP handler on 192.168.43.48:4444 
[-] Module dependencies (impacket) missing, cannot continue
[-] Exploit aborted due to failure: unknown: Module exited abnormally
[*] Exploit completed, but no session was created.

but after adding this exception of yours,  it show like this :

[*] Started reverse TCP handler on 192.168.43.48:4444 
[-] Exploit aborted due to failure: unknown: Module exited abnormally
[*] Exploit completed, but no session was created.

-----------------------------------------------------------------------------------------------------------------------

am i missing something, if yes please correct me. coz i need this exploit to complete the tryhackme room name 'eternal blue' and i'm already too late coz of laziness and delays :-1: 

![Screenshot at 2020-07-23 23-41-09](https://user-images.githubusercontent.com/60420716/88322441-5969e500-cd3e-11ea-82ca-855e03a0aa17.png)

again, correct me if I've done something wrong here ...

thanks for all you contributions and helping!

Reply:
@rootz491, This fix has been in Metasploit for two years, including every 5.x release. The the file in the screenshot provided won't run at all, as it is not a valid Python program. Please use an up-to-date version of Metasploit and create a new issue for things you encounter.